agent/consul/server.go

```diff
@@ -968,6 +968,8 @@ func (s *Server) Shutdown() error {
 		s.config.NotifyShutdown()
 	}
 
+	s.fsm.State().Abandon()
```

Comment:
Calling this out as, in case there are implications I have missed. Currently `Abandon` is only called at the end of a restore. This needs to be called to stop `EventPublisher` so that the background goroutine is shutdown.  From what I can, previously this is only used by consumers of the state store to learn that the store has been abandoned, so that a `WatchSet` can stop blocking. So this seems safe to call here.

Reply:
I realised there is a problem with this in a comment below.